src/jobflow_remote/jobs/daemon.py

```diff
@@ -405,61 +418,104 @@ def start(
+                    f"- hostname: {drr['hostname']}\n"
+                    f"- project_name: {drr['project_name']}\n"
+                    f"- start_time: {drr['start_time']}\n"
+                    f"- last_pinged: {drr['last_pinged']}\n"
+                    f"- daemon_dir: {drr['daemon_dir']}\n"
+                    f"- user: {drr['user']}"
+                )
 
-        if error is not None:
-            if raise_on_error:
-                raise DaemonError(error)
-            logger.error(error)
-            return False
-        return True
+            elif status == DaemonStatus.SHUT_DOWN:
+                error = self.start_supervisord(
+                    num_procs_transfer=num_procs_transfer,
+                    num_procs_complete=num_procs_complete,
+                    single=single,
+                    log_level=log_level,
+                    connect_interactive=connect_interactive,
+                )
+            elif status == DaemonStatus.STOPPED:
+                self.shut_down(raise_on_error=raise_on_error)
+                error = self.start_supervisord(
+                    num_procs_transfer=num_procs_transfer,
+                    num_procs_complete=num_procs_complete,
+                    single=single,
+                    log_level=log_level,
+                    connect_interactive=connect_interactive,
+                )
+                # else:
+                #     error = self.start_processes()
+            elif status == DaemonStatus.STOPPING:
+                error = "Daemon process are stopping. Cannot start."
+            else:
+                error = f"Daemon status {status} could not be handled"
+
+            if error is not None:
+                if raise_on_error:
+                    raise DaemonError(error)
+                logger.error(error)
+                return False
+            lock.update_on_release = {
+                "$set": {"running_runner": self._get_runner_info()}
+            }
+            return True
 
     def stop(self, wait: bool = False, raise_on_error: bool = False) -> bool:
-        status = self.check_status()
-        if status in (
-            DaemonStatus.STOPPED,
-            DaemonStatus.STOPPING,
-            DaemonStatus.SHUT_DOWN,
-        ):
-            return True
+        db_filter = {"running_runner": {"$exists": True}}
+        with self.job_controller.lock_auxiliary(filter=db_filter) as lock:
+            if lock is None:
+                # print('Handle case where the document is not present!')
+                pass
+            if lock.is_locked:
+                # print('Handle case where the document is locked')
+                pass
+            status = self.check_status()
+            if status in (
+                DaemonStatus.STOPPED,
+                DaemonStatus.STOPPING,
+                DaemonStatus.SHUT_DOWN,
+            ):
+                lock.update_on_release = {"$set": {"running_runner": None}}
+                return True
```

Comment:
Indeed ... I guess it should only set it to None if the status was not stopped before. Or I should make a matching between the information in the running runner and the information about the runner on the machine ?

Reply:
I don't know. What happens if I have only one machine that runs the Runner (which is likely 99% of the use cases) . It is started but the machine goes down abruptly (e.g. power cut). The document will not be updated but the deamon is shut down. What happens then? How am I supposed to deal with that situation? I likely don't want to go through an annoying procedure of having to reset the document manually before I can restart the db. Could it make sense that there is a global check that whenever the runner is not active and the document matches the local information it will clean up the content of the document?

Acutally this is also a linked to the `start` method works. Currently in a scenario as the one I just described I would never be able to start the runner again, right?